packages/calypso-e2e/src/lib/utils/validate-translations.ts

```diff
@@ -0,0 +1,121 @@
+import { sprintf } from '@wordpress/i18n';
+import { Page, ElementHandle } from 'playwright';
+
+const GLOTPRESS_ORIGINALS_ENDPOINT =
+	'https://translate.wordpress.com/api/translations/-query-by-originals';
+const GLOTPRESS_PROJECT = 'wpcom';
+
+const selectors = {
+	translatableElement: '[data-e2e-string]',
+};
+
+type OriginalString = { singular: string; plural?: string; context?: string };
+type Translation = {
+	original: OriginalString;
+	translations: {
+		translation_0: string;
+	}[];
+};
+
+/**
+ * Fetch translations for originals.
+ */
+async function fetchTranslations(
+	page: Page,
+	originals: OriginalString[],
+	locale: string
+): Promise< Translation[] | null > {
+	const params = {
+		url: GLOTPRESS_ORIGINALS_ENDPOINT,
+		data: {
+			project: GLOTPRESS_PROJECT,
+			locale_slug: locale,
+			original_strings: JSON.stringify( originals ),
+		},
+	};
+
+	return page.evaluate( async ( { url, data } ) => {
+		const body = new URLSearchParams();
+		for ( const key in data ) {
+			body.append( key, data[ key as 'project' | 'locale_slug' | 'original_strings' ] );
+		}
+
+		return window
+			.fetch( url, {
+				method: 'POST',
+				headers: {
+					'content-type': 'application/x-www-form-urlencoded',
+				},
+				body: body.toString(),
+			} )
+			.then( ( res ) => res.json() )
+			.then( ( res ) => {
+				delete res.originals_not_found;
+				return res;
+			} )
+			.catch( () => null );
+	}, params );
+}
+
+/**
+ * Get elements translations.
+ */
+async function getElementsTranslations(
+	page: Page,
+	elements: ElementHandle[],
+	locale: string
+): Promise< Translation[] | null > {
+	// Default locale doesn't have translations
+	if ( locale === 'en' ) {
+		return null;
+	}
+
+	const originals: OriginalString[] = await Promise.all(
+		elements.map( async ( element ) => {
+			const singular = ( await element.getAttribute( 'data-e2e-string' ) ) || '';
```

Comment:
> Just to clarify, elements that don't have a value in data-e2e-string aren't really translatable, correct?
That's correct. We don't need to apply any filtering, because the [translatableElement](https://github.com/Automattic/wp-calypso/pull/57707/commits/4672e76befb6a1f561b7b8ca8ba7ff8849101301#diff-c9f4039cb8be26088b4672860ea55820f2f13bbc08555f87923c9b181c2a43dbR11) is already targeting elements with `data-e2e-string` attribute. The empty string fallback value here was only needed to fulfil OriginalString type def, but I ended up changing it slightly so it's no longer needed.